src/plot.cpp

```diff
@@ -1,10 +1,1010 @@
+      err_msg << "Mutliple meshlines specified in plot " << id;
+      fatal_error(err_msg);
+    }
+  }
+}
+
+void
+ObjectPlot::set_mask()
+{
+  // Deal with masks
+  std::vector<pugi::xml_node> mask_nodes;
+  mask_nodes = get_child_nodes(_plot_node, "mask");
+  int n_masks = mask_nodes.size();
+
+  if (n_masks > 0) {
+    if (PLOT_TYPE::VOXEL == type) {
+      if (openmc_master) {
+        std::stringstream wrn_msg;
+        wrn_msg << "Mask ignored in voxel plot " << id;
+        warning(wrn_msg);
+      }
+    }
+
+    if (1 == n_masks) {
+      // Get pointer to mask
+      pugi::xml_node mask_node = mask_nodes[0];
+
+      // Determine how many components there are and allocate
+      int n_comp;
+      n_comp = node_word_count(mask_node, "components");
+      if (0 == n_comp) {
+        std::stringstream err_msg;
+        err_msg << "Missing <components> in mask of plot " << id;
+        fatal_error(err_msg);
+      }
+      std::vector<int> iarray = get_node_array<int>(mask_node, "components");
+
+      // First we need to change the user-specified identifiers to indices
+      // in the cell and material arrays
+      int col_id;
+      for (int j = 0; j < iarray.size(); j++) {
+        col_id = iarray[j];
+
+        if (PLOT_COLOR_BY::CELLS == color_by) {
+          if (cell_map.find(col_id) != cell_map.end()) {
+            iarray[j] = cell_map[col_id];
+          }
+          else {
+            std::stringstream err_msg;
+            err_msg << "Could not find cell " << col_id
+                    << " specified in the mask in plot " << id;
+            fatal_error(err_msg);
+          }
+        } else if (PLOT_COLOR_BY::MATS == color_by) {
+          if (material_map.find(col_id) != material_map.end()) {
+            iarray[j] = material_map[col_id];
+          }
+          else {
+            std::stringstream err_msg;
+            err_msg << "Could not find material " << col_id
+                    << " specified in the mask in plot " << id;
+            fatal_error(err_msg);
+          }
+        }
+      }
+
+      // Alter colors based on mask information
+      for(int j = 0; j < colors.size(); j++) {
+        if (std::find(iarray.begin(), iarray.end(), j) == iarray.end()) {
+          if (check_for_node(mask_node, "background")) {
+            std::vector<int> bg_rgb = get_node_array<int>(mask_node, "background");
+            colors[j]->rgb[RED] = bg_rgb[RED];
+            colors[j]->rgb[GREEN] = bg_rgb[GREEN];
+            colors[j]->rgb[BLUE] = bg_rgb[BLUE];
+          } else {
+            colors[j]->rgb[RED] = 255;
+            colors[j]->rgb[GREEN] = 255;
+            colors[j]->rgb[BLUE] = 255;
+          }
+        }
+      }
+
+    } else {
+      std::stringstream err_msg;
+      err_msg << "Mutliple masks specified in plot " << id;
+      fatal_error(err_msg);
+    }
+  }
+}
+
+ObjectPlot::ObjectPlot(pugi::xml_node plot_node):
+index_meshlines_mesh(-1)
+{
+  _plot_node = plot_node;
```

Comment:
Hrmm yeah I was torn between what I've done and the repetitive-feeling method signatures involved passing that around. Having that extra space around probably isn't great though. If it were valid after initialization it might make more sense, but I think you're right. Will update. Thanks!